pkg/proc/target_exec.go

```diff
@@ -125,7 +125,7 @@ func (dbp *Target) Continue() error {
 			case loc.Fn.Name == "runtime.breakpoint":
 				// In linux-arm64, PtraceSingleStep seems cannot step over BRK instruction
 				// (linux-arm64 feature or kernel bug maybe).
-				if !arch.BreakInstrMovesPC() {
```

Comment:
You should make BreakInstrMovesPC return true if you don't want this for "arm"

Reply:
Golang use UND instruction `0xf0, 0x01, 0xf0, 0xe7` to act as break instruction, this instruction will cause ptrace to stop at the PC of current UND instruction, so BreakInstrMovesPC should be false I think, and as we do implementation the singleStep, we are able to move PC no matter what instruction it is, so I change there to compare with `BreakInstrMovesPC` to compare with `arm64` arch, as the comment here said `PtraceSingleStep seems cannot step over BRK instruction // (linux-arm64 feature or kernel bug maybe).`

Reply:
In that case I think you should revert this change to check for BreakInstrMovesPC and then remove the special code in resolvePCs and singleStep to NOP breakpoint instructions.

Reply:
Got, I change it back to check against BreakInstrMovesPC , thanks